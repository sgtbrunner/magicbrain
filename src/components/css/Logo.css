.Tilt {
  	background-image: radial-gradient(circle, rgb(255, 140, 20) 20%, rgb(150, 90, 250) 90%);
	height: 150px;
	width: 150px;
	border-radius: 100px;
}

.tiltbox {
	padding-top: 25px;
	border-radius: 100px;
}