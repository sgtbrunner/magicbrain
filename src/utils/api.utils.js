const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
