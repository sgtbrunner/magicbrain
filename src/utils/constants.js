export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
