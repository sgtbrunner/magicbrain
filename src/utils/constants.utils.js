const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
