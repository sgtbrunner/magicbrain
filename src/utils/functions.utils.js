export const getUserFromLocalStorage = () => localStorage.getItem('user');
