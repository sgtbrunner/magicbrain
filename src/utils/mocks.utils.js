export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
