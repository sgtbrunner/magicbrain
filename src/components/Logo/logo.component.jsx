import React from 'react';
import Tilt from 'react-tilt';
import brain from '../../assets/images/brain.png';
import './logo.styles.css';

const Logo = () => {
  return (
    <div className="ma4 mt0 tiltbox centered">
      <Tilt className="Tilt shadow-2" options={{ max: 100 }} style={{ height: 130, width: 130 }}>
        <div className="Tilt-inner pa4 tiltbox">
          <img className="noselect" src={brain} alt="logo" />
        </div>
      </Tilt>
    </div>
  );
};

export default Logo;
