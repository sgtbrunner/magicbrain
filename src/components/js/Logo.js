import React from 'react';
import Tilt from 'react-tilt';
import brain from '../../img/brain.png';
import '../css/Logo.css'

const Logo = () => {
	return (
		<div className="ma4 mt0 tiltbox centered">
			<Tilt className="Tilt shadow-2" options={{ max : 100 }} style={{ height: 130, width: 130 }} >
				 <div className="Tilt-inner pa4 tiltbox">
				 	<img src={brain} alt="logo"/>
				 </div>
			</Tilt>
		</div>
	)
}

export default Logo;