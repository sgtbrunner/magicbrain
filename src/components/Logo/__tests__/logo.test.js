import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Logo from '..';

const mountLogoComponent = () => mountComponent({ component: Logo });

describe('Logo component', () => {
  it('should match snapshot', () => {
    const { container } = mountLogoComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render logo image within component', () => {
    mountLogoComponent();

    const logo = screen.getByRole('img');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveProperty('alt', 'logo');
  });
});
