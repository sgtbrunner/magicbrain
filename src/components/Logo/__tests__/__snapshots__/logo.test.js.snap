// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should match snapshot 1`] = `
<div>
  <div
    class="ma4 mt0 tiltbox centered"
  >
    <div
      class="Tilt shadow-2"
      style="height: 130px; width: 130px;"
    >
      <div
        class="Tilt-inner pa4 tiltbox"
      >
        <img
          alt="logo"
          class="noselect"
          src="brain.png"
        />
      </div>
    </div>
  </div>
</div>
`;
